dali/operators/decoder/audio/audio_decoder.h

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_DECODER_AUDIO_AUDIO_DECODER_H_
+#define DALI_OPERATORS_DECODER_AUDIO_AUDIO_DECODER_H_
+
+#include <memory>
+#include "dali/core/span.h"
+
+namespace dali {
+
+struct decoder_runtime_error : public DALIException {
+  decoder_runtime_error(const std::string &err) : DALIException(err) {}  // NOLINT
+};
+
+struct AudioMetadata {
+  int length;
+  int sample_rate;  /// [Hz]
```

Comment:
I'm wondering if it's going to bite us - in principle, frequency is a real number. On the other hand, I'd yet have to see a sound format with non-integer-Hz sampling rate.

Reply:
After discussion, we decided to keep in `int` and see what happens in future